bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/DbLedgerStorage.java

```diff
@@ -74,10 +83,121 @@
  */
 public class DbLedgerStorage implements CompactableLedgerStorage {
 
+    /**
+     * This class borrows the logic from FileInfo.
+     *
+     * <p>This class is used for holding all the transient states for a given ledger.
+     */
+    private static class TransientLedgerInfo
+            extends Watchable<LastAddConfirmedUpdateNotification>
+            implements AutoCloseable {
+
+        // lac
+        private Long lac = null;
```

Comment:
use LedgerHandle.INVALID_ENTRY_ID and you'll avoid the autoboxing and all the null checks.

Reply:
null has a different meaning than INVALID_ENTRY_ID, when it is null, we need to read the last entry to examine the lac.

even we can work this around to use long, would still prefer keeping this consistent with the implementation in SortedLedgerStorage, which is safer.

Reply:
null only has a different meaning because you're loading uninitialized entries into the cache. I'd do it differently, but won't block on this.

Reply:
that means you need to refactor the whole logic, not just ledger storage, which I will hesitate to do it.